crates/ark/src/lsp/diagnostics.rs

```diff
@@ -804,6 +819,7 @@ fn recurse_call_arguments_default(
         }
     }
 
+    context.in_call = in_call;
```

Comment:
The `?` within the `children` loop mean that we might not set `in_call` on the way out.

I recognize that if an error occurs we will basically refuse to do diagnostics all together here, so it doesn't matter _too_ much, but I really had to think about it to figure that out.

Is it possible to move the actual recursion into a sub helper that we call from here? We'd:
- Set `context.in_call` to new value
- Call the sub helper, save value as `out` (a `Result`)
- Set `context.in_call` to old value
- Return `out`

Or create a helper function that does this called `with_in_call(f: FnOnce)`

That way we can't "forget" to reset `in_call`, in case in the future we do some kind of error recovery when we hit an error rather than bailing completely on diagnostics as we do now

Reply:
oh good point, I thought explicitly about absence of exceptions, but somehow forgot about early returns -_-